### PR TITLE
Blog App -validations and model specs.

### PR DESCRIPTION
**Pull request by MohammadYaser**
Pull Request: Model Validations and Unit Specs Addition 🧪✨
Changes Made
Added validations to enhance data integrity for the User and Post models. 🛡️
Implemented unit specs for all models' methods and validations. 🧪
Model Validations
User Model
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
Post Model
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Unit Specs
Implemented unit tests for all models' methods and validations.
Ensured that validations are working as expected.
Validations added to prevent invalid data entry and ensure data consistency.
Unit specs provide comprehensive coverage for models' behavior.
Testing
Ran unit specs to validate model methods and ensure they are error-free.
Manually tested validations in the 'rails console' to verify their functionality.
Next Steps
 Collaborate with team members for code review.
 Address any feedback received during the review process.
 Ensure all tests pass successfully.
completed

- [ ]  Validations added for User and Post models.
- [ ]  Unit specs implemented for all models' methods and validations.
- [ ]  README updated with relevant information.

Please review and allow merge.

Thank you! 🙌

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  it 'updates the posts counter for a my_user' do
+    my_user = User.create(name: 'yaser', post_counter: 0)
+    post = my_user.posts.create(title: 'My Post', comments_counter: 0, likes_counter: 0)
+
+    post.increment_user_post_counter
+
+    my_user.reload
+    expect(my_user.post_counter).to eq(2)
+  end
```
Comment (DammyShittu) [inline spec/models/post_spec.rb]: - [ ] Kindly add test for `recent_comments` method.

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+RSpec.describe Like, type: :model do
+  let(:user) { create(:user) }
+  let(:post) { create(:post) }
+
+  describe 'associations' do
+    it { should belong_to(:user) }
+    it { should belong_to(:post) }
+  end
+end
```
Comment (DammyShittu) [inline spec/models/like_spec.rb]: - [ ] Kindly add test for `increment_post_likes_counter` method.

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  describe 'validations' do
+    it { should validate_presence_of(:text) }
+  end
+
+  describe 'associations' do
+    it { should belong_to(:user) }
+    it { should belong_to(:post) }
+  end
+end
```
Comment (DammyShittu) [inline spec/models/comment_spec.rb]: - [ ] Kindly add test for the `increment_post_comments_counter` method in the Comment model file.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @MohammadYaser,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @MohammadYaser, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] All requested changes are made ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**